packages/node_modules/@webex/internal-plugin-calendar/src/calendar.js

```diff
@@ -4,9 +4,70 @@
 
 import {WebexPlugin} from '@webex/webex-core';
 
+import CalendarCollection from './collection';
+
 const Calendar = WebexPlugin.extend({
   namespace: 'Calendar',
 
+  register() {
+    // Calendar mercury events listener
+    this.webex.internal.mercury.on('event:calendar.meeting.create', (envelope) => {
+      this._handleCreate(envelope.data);
+    });
+    this.webex.internal.mercury.on('event:calendar.meeting.update', (envelope) => {
+      this._handleUpdate(envelope.data);
+    });
+    this.webex.internal.mercury.on('event:calendar.meeting.create.minimal', (envelope) => {
+      this._handleCreate(envelope.data);
+    });
+    this.webex.internal.mercury.on('event:calendar.meeting.update.minimal', (envelope) => {
+      this._handleUpdate(envelope.data);
+    });
+    this.webex.internal.mercury.on('event:calendar.meeting.delete', (envelope) => {
+      this._handleDelete(envelope.data);
+    });
+  },
+
+  _handleUpdate(data) {
+    const id = CalendarCollection.set(data.calendarMeetingExternal);
+
+    this.trigger('calendar:meeting:update', CalendarCollection.get(id));
+  },
+
+  _handleCreate(data) {
+    const id = CalendarCollection.set(data.calendarMeetingExternal);
+
+    this.trigger('calendar:meeting:create', CalendarCollection.get(id));
+  },
+
+  _handleDelete(data) {
+    const item = CalendarCollection.remove(data.calendarMeetingExternal.id);
+
+    this.trigger('calendar:meeting:delete', item);
+  },
+
+  // default to yesterday to end of the week so 8 days total
+  populate() {
```

Comment:
What do yo think about allowing properties to be passed in to the method to override the default config? If that's too much flexibility that's fine

Reply:
do you need to return this.list()?

Reply:
this should be done at startup time, and we re format to an object, unnecessary to return

i think I'm cool with properties